py-polars/polars/series/datetime.py

```diff
@@ -944,6 +944,82 @@ def nanosecond(self) -> Series:
 
         """
 
+    def minute_of_day(self) -> Series:
+        """
+        Extract the minute of the day from the underlying DateTime representation.
+
+        Applies to Datetime columns.
+
+        Returns the minute number from 0 to 1439.
+
+        Returns
+        -------
+        Series
+            Series of data type :class:`UInt32`.
+
+        Examples
+        --------
+        >>> from datetime import datetime
+        >>> start = datetime(2001, 1, 1)
+        >>> stop = datetime(2001, 1, 1, 1, 4, 0)
+        >>> date = pl.date_range(start, stop, interval="2m", eager=True)
+        >>> date
+        shape: (3,)
+        Series: 'date' [datetime[μs]]
+        [
+                2001-01-01 00:00:00
+                2001-01-01 00:02:00
+                2001-01-01 00:04:00
+        ]
+        >>> date.dt.minute_of_day()
+        shape: (3,)
+        Series: 'date' [u32]
+        [
+                0
+                2
+                64
+        ]
+
+        """
+
+    def second_of_day(self) -> Series:
+        """
+        Extract the second of the day from the underlying DateTime representation.
+
+        Applies to Datetime columns.
+
+        Returns the second number from 0 to 86399.
+
+        Returns
+        -------
+        Series
+            Series of data type :class:`UInt32`.
+
+        Examples
+        --------
+        >>> from datetime import datetime
+        >>> start = datetime(2001, 1, 1)
+        >>> stop = datetime(2001, 1, 1, 1, 4, 0)
+        >>> date = pl.date_range(start, stop, interval="2m", eager=True)
+        >>> date
+        shape: (3,)
+        Series: 'date' [datetime[μs]]
+        [
+                2001-01-01 00:00:10
+                2001-01-01 00:02:20
+                2001-01-01 01:00:00
+        ]
+        >>> date.dt.second_of_day()
+        shape: (3,)
+        Series: 'date' [u32]
+        [
+                10
+                1440
+                3600
+        ]
```

Comment:
is this not already possible (and efficient) with
```python
In [20]: date.dt.second() + date.dt.minute()*60 + date.dt.hour()*3600
Out[20]:
shape: (3,)
Series: 'date' [u32]
[
        10
        140
        3600
]
```
?

In addition to being concerned about "API bloat", I think this is a bit more explicit - is "second of day" a standard term? To me at least it's not obvious what it means, just by reading it

Reply:
I agree that it is possible and efficient but I think it is a common enough operation that a short hand function could be useful. Also it makes it easier when the date/ time is parsed directly from a string e.g
```python
time_string.str.strptime(pl.Time, "%H:%M:%S").dt.second_of_day()
```
>is "second of day" a standard term?
Not that I know of but happy to hear other suggestions